.github/workflows/ci.yml

```diff
@@ -20,25 +20,35 @@ on: [pull_request]
 jobs:
   build:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go-version: [ '1.20', '1.21', '1.22' ]
     steps:
       - uses: actions/checkout@v3
-      - uses: actions/setup-go@v3
+      - uses: actions/setup-go@v5
+        with:
+          go-version: ${{ matrix.go-version }}
       - run: make build
 
   lint:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
+      - uses: actions/setup-go@v5
+        with:
+          go-version: '1.20'
       - name: Check license header
         run: docker run --rm -v $(pwd):/github/workspace ghcr.io/korandoru/hawkeye-native:v3 check
       - name: Run golangci-lint
-        run: make lint
+        uses: golangci/golangci-lint-action@v6
+        with:
+          version: v1.51.2
```

Comment:
Why this change? We still provide the Makefile target. This unaligned may silently cause CI logic and dev logic different ..

Both download binaries so that time consumption should be similiar.

Reply:
golangci/golangci-lint-action can report lint result to the ci summary.

When we update the golangci-lint, we must update the workflow and makefile files to keep same version.